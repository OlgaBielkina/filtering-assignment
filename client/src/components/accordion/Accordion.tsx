import React, { useState } from 'react';

interface Props {
    children: React.ReactNode;
}

export interface IAccordionContext {
    expanded: boolean;
    handleExpanded?: () => void;
}

export const AccordionContext = React.createContext<IAccordionContext>({
    expanded: false,
    handleExpanded: undefined,
});

export const Accordion: React.FC<Props> = ({ children }) => {
    const [expanded, setExpanded] = useState(false);
    const handleExpanded = () => setExpanded(!expanded);
    const data = { expanded, handleExpanded };

    return <AccordionContext.Provider value={data}>{children}</AccordionContext.Provider>;
};
