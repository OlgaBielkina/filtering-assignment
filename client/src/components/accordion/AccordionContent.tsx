import React, { useContext } from 'react';
import { AccordionContext } from './Accordion';
import styles from './AccordionContent.module.scss';

interface Props {
    children: React.ReactNode;
}

export const AccordionContent: React.FC<Props> = ({ children }) => {
    const context = useContext(AccordionContext);

    return (
        <div data-testid="accordionContent" className={context.expanded ? '' : styles.hidden}>
            {children}
        </div>
    );
};
