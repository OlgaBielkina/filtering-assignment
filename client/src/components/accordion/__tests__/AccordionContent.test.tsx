import React from 'react';
import { render } from '@testing-library/react';
import { AccordionContent } from '../AccordionContent';
import { AccordionContext, IAccordionContext } from '../Accordion';

describe('AccordionContent', () => {
    const providerData: IAccordionContext = {
        expanded: false,
        handleExpanded: jest.fn(),
    };

    it('should render collapsed content', () => {
        const { asFragment, getByText } = render(
            <AccordionContext.Provider value={providerData}>
                <AccordionContent>test text</AccordionContent>
            </AccordionContext.Provider>,
        );

        expect(getByText('test text')).toHaveClass('hidden');
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render expanded content', () => {
        const { asFragment, getByText } = render(
            <AccordionContext.Provider value={{ ...providerData, expanded: true }}>
                <AccordionContent>test text</AccordionContent>
            </AccordionContext.Provider>,
        );

        expect(getByText('test text')).not.toHaveClass('hidden');
        expect(asFragment()).toMatchSnapshot();
    });
});
