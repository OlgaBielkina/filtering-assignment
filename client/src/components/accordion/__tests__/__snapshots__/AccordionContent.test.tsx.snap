// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionContent should render collapsed content 1`] = `
<DocumentFragment>
  <div
    class="hidden"
    data-testid="accordionContent"
  >
    test text
  </div>
</DocumentFragment>
`;

exports[`AccordionContent should render expanded content 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="accordionContent"
  >
    test text
  </div>
</DocumentFragment>
`;
